classes/class-sensei-teacher.php

```diff
@@ -1405,4 +1412,69 @@ public function limit_teacher_edit_screen_post_types( $wp_query ) {
 
     } // end limit_teacher_edit_screen_post_types()
 
+
+    /**
+     * Sensei_Teacher::restrict_posts_menu_page()
+     *
+     * Remove the Posts menu page for teachers and restrict access to it.
+     * We have to do this because we give teachers the 'edit_posts' cap
+     * so they can 'moderate_comments' as well.
+     *
+     * @since 1.8.7
+     * @access public
+     * @parameters void
+     * @return void
+     */
+
+    public function restrict_posts_menu_page() {
+
+        global $pagenow, $typenow;
+
+        $user = wp_get_current_user();
+
+        if ( in_array( 'teacher', (array) $user->roles ) ) {
```

Comment:
@mahangu while thinking through this again, another scenario came up:
 
What happens in the case where a user who already has access to the edit menu (say author / contributor) is given the teacher role. This will mean that that user will be denied what he already has access to. We will need to check if the user doesn't have other roles that allows them access as simply checking the edit_posts will not work.